packages/components/src/Notification/ToastNotification/types.ts

```diff
@@ -0,0 +1,33 @@
+import { NotificationType } from "../types"
+
+type Modify<T, R> = Omit<T, keyof R> & R
+
+export type ToastNotification = {
+  id: string
+  type: NotificationType
+  title: string
+  message: React.ReactNode
+  onHide?: () => void
+  /**
+   * Removes the dismiss trigger. If this is true you will need to manage the removal of notifications manually.
+   * @default false
+   */
+  persistent?: boolean
+} & {
+  [key: `data-${string}`]: unknown
```

Comment:
The original component took automationId as a prop, which was being used to pass it in as a test id, which felt very out of date with modern practice. I figured pulling in data attribution in a `restProps` was a better way tp pass test ids into the component

Reply:
We have a `DataAttributes` type which helps with this 😉 That being said, if we aren't extending HTML attributes and spreading the props, it might be safer in the meantime to just do what G has been doing and only add `data-testid?: string` as a prop to replace `automationId`